src/Ocelot/Configuration/QoSOptions.cs

```diff
@@ -32,37 +32,83 @@ public QoSOptions(
             TimeoutValue = timeoutValue;
         }
 
+        public QoSOptions(
+            int exceptionsAllowedBeforeBreaking,
+            int durationOfBreak,
+            double failureRatio,
+            int timeoutValue,
+            string key)
+        {
+            DurationOfBreak = durationOfBreak;
+            ExceptionsAllowedBeforeBreaking = exceptionsAllowedBeforeBreaking;
+            Key = key;
+            TimeoutValue = timeoutValue;
+            FailureRatio = failureRatio;
+        }
+
+        public QoSOptions(
```

Comment:
Why did you add both constructors? I don't see any changes in the Creators' logic.
It appears these constructors were added solely for testing purposes. I would suggest switching to `QoSOptionsBuilder` and removing these two constructors.
Therefore, you will need to update `QoSOptionsBuilder` by adding new initialization methods.